### PR TITLE
AP_Scripting:provide altitude loss safety abort for plane aerobatics

### PR DESCRIPTION
**Pull request by Hwurzburg**
would have prevented a crash I had today while distracted running the Airshow as a demo to someone...duh!
my fault but this would have saved the plane

### CODE REVIEW

**Comment by tridge** [comment]
I think using (current_measured_pos_ef - path_var.start_pos):z() this should be distance below the start pos

**Comment by Hwurzburg** [comment]
done...thanks...will SITL this weekend

**Comment by Hwurzburg** [comment]
> I think using (current_measured_pos_ef - path_var.start_pos):z() this should be distance below the start pos

@tridge think one is a 3d vector and one is a location....so have to modify that somehow....but over my head

File: libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua [hunk 1]
```diff
@@ -2173,6 +2174,11 @@ function do_path()
       last_named_float_t = now
       gcs:send_named_float("PERR", pos_error_ef:length())
    end
+   
+   if ((current_measured_pos_ef - path_var.start_pos):z()) > AEROM_ALT_ABORT.get() then
```
Comment (magicrub) [inline libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua:2178]: this is supposed to be ```:get()```, right?

Comment (magicrub) [reply libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua:2178]: I think maybe the > is backwards?
Also, this is probably a lot easier to think about if it was this:
```suggestion
   if ((current_measured_pos_ef:z() - path_var.start_pos:z()) < AEROM_ALT_ABORT:get() then
```

Comment (Hwurzburg) [reply libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua]: right...but I was using the wrong variables also...fixed and tested now

**Comment by Hwurzburg** [comment]
> I think using (current_measured_pos_ef - path_var.start_pos):z() this should be distance below the start pos

used different variable in same coordinate system....works like a champ now...

File: libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua [hunk 2]
```diff
@@ -42,6 +42,7 @@ AEROM_STALL_PIT = bind_add_param('STALL_PIT', 24, -20)
 -- 25 was AEROM_KE_TC
 AEROM_KE_RUDD = bind_add_param('KE_RUDD', 26, 25)
 AEROM_KE_RUDD_LK = bind_add_param('KE_RUDD_LK', 27, 0.25)
+AEROM_ALT_ABORT = bind_add_param('PERR_ABORT',28,15)
```
Comment (tridge) [inline libraries/AP_Scripting/applets/Aerobatics/FixedWing/plane_aerobatics.lua]: wrong param name, should be ALT_ABORT

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**